### PR TITLE
Changed-type-mapping

### PR DESCRIPTION
**Pull request by MoizMasud**

### CODE REVIEW

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Review by Akshat55 — APPROVED** [review]

File: src/fragment-components/a-component.tsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ const dropStyleAfter = cx(dropStyle, css`
 `);
 
 export interface ComponentInfo {
+	type: any;
```
Comment (zvonimirfras) [inline src/fragment-components/a-component.tsx:54]: Isn't type string?
```suggestion
	type: string;
```

File: src/fragment-components/accordion/a-accordion-item.tsx [hunk 2]
```diff
@@ -106,6 +106,7 @@ export const componentInfo: ComponentInfo = {
 	</AAccordionItem>,
 	keywords: ['accordion', 'item'],
 	name: 'Accordion Item',
```
Comment (zvonimirfras) [inline src/fragment-components/accordion/a-accordion-item.tsx:108]:
```suggestion
	name: 'Accordion item',
```

File: src/utils/fragment-tools.ts [hunk 3]
```diff
@@ -65,6 +65,7 @@ export const getAllComponentStyleClasses = (componentObj: any, fragments: any[])
 
 			styleClasses = {
 				...styleClasses,
+				// eslint-disable-next-line @typescript-eslint/no-use-before-define
```
Comment (zvonimirfras) [inline src/utils/fragment-tools.ts:68]: This has a smell... Why is this needed?
What other way do we have for solving this?

**Review by zvonimirfras — CHANGES_REQUESTED** [review]

File: src/utils/fragment-tools.ts (hunk 3 above)
Comment (MoizMasud) [reply src/utils/fragment-tools.ts:68]: 'getAllFragmentStyleClasses' was used before it was defined. It still finds the function, but eslint doesnt like it so i disabled the rule for that line.

Comment (zvonimirfras) [reply src/utils/fragment-tools.ts:68]: Right, and if we change the order, the other one is gonna be problematic...

I found a way to make it go away (declare it before, then reassign it, then export), but I'm not sure it's cleaner...

```suggestion
				// we can't avoid this without a messy declare+reassign+export
				// eslint-disable-next-line @typescript-eslint/no-use-before-define
```

**Review by zvonimirfras — APPROVED** [review]
